Syntaxes/References.sublime-syntax

```diff
@@ -1,34 +1,47 @@
 %YAML 1.2
 ---
+# [Subl]: https://www.sublimetext.com/docs/3/syntax.html
+# [LSP]: https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md
 hidden: true
 scope: output.lsp.references
 
+variables:
+  start_of_reference_body: ^\s+(?=\d)
+  filename_and_colon: ^\s*(\S)\s+(.*)(:)$
+
 contexts:
   main:
     - include: references-preamble
     - include: references-body
 
   references-preamble:
-    # References to "function" at path\file.py:
-    - match: ^(.*)(:)$
+    - match: '{{filename_and_colon}}'
       captures:
-        0: meta.references.preamble.lsp
-        1: string.unquoted.lsp
-        2: punctuation.separator.lsp
+        0: meta.reference.preamble.lsp
+        1: punctuation.section.references.preample.lsp
+        2: string.unquoted.lsp
+        3: punctuation.separator.lsp
 
   references-body:
-    # Must match the
-    #    - "result_file_regex" setting
-    #    - reference output format
-    # Examples:
-    #    ◌ c:\path name\file name.py 423:32
-    #    ◌ /unixpath/file.py 815:1
-    - match: ^\s+(\S)\s+((\S.+)\s+(\d+)(?:(:)(\d+))?)$
+    - match: '{{start_of_reference_body}}'
+      push:
+        - ensure-reference-meta-scope
```

Comment:
Since you already have the multi-push, you might as well go ahead and highlight any following referenced code as `source` (unspecified which one as we don't know). Some color schemes, like mine, have background color highlighting for inline code segments on `source source`, for example.

I.e.
```yaml
  references-body:
    - match: '{{start_of_reference_body}}'
      push:
        - ensure-reference-meta-scope
        - code
        - expect-line-maybe-column
      
  code:
    - match: '(?=\S)'
      push:
        - meta_scope: source
        - include: pop-at-end
```

Reply:
Thanks for the instructions. 
![2018-09-21-132934_683x564_scrot](https://user-images.githubusercontent.com/22029477/45878814-b68f5b00-bda2-11e8-9f19-92cc3f198e1a.png)
But how do I stop the scope `source` from spanning to the next line?

```
          38:13   file_path = uri_to_filename(location.get("uri"))
	  40:47   file_location = "{}:{}:{}".format(file_path, start.row + 1, start.col + 1)
  scopes  ^ output.lsp.references meta.reference.body.lsp source

Reply:
Ah, my bad. You need to change toe "push" to a "set", otherwise you won't be able to leave it.